src/renderer/components/ft-toast/ft-toast.js

```diff
@@ -0,0 +1,54 @@
+import Vue from 'vue'
+import FtToastEvents from './ft-toast-events.js'
+
+export default Vue.extend({
+  name: 'FtToast',
+  data: function () {
+    return {
+      toasts: [
+      ],
+    }
+  },
+  mounted: function () {
+    FtToastEvents.$on('toast.open', this.open)
+  },
+  beforeDestroy: function () {
+    FtToastEvents.$off('toast.open', this.open)
+  },
+  methods: {
+    performAction: function (toast) {
```

Comment:
Let's change this to accept the index of the object instead of the object itself.  Also instead of calling the close function, let's just remove it from the array in here.  This will prevent a gap in the toast messages when clicked on while not affecting the rest of them when gracefully closing.  This prevents the toast from fading away when clicked on however I feel like this is okay.

```
performAction: function (index) {
      this.toasts[index].action()
      this.toasts.splice(index, 1)
}
```